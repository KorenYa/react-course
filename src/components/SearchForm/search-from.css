.search-form {
    padding: 20px 60px;
    background: url('https://assets.nflxext.com/ffe/siteui/vlv3/a27b8244-8c0b-4db4-984e-0cd57daa489d/8e38e13b-8cfc-410e-8d2d-a6c9e717d9b8/UA-en-20190506-popsignuptwoweeks-perspective_alpha_website_large.jpg') center center;
    color: #ffffff;
}

.search-form__title {
    color: #f55263;
    font-weight: 600;
    margin-bottom: 30px;
}

.search-form__subtitle {
    text-transform: uppercase;
    font-size: 14px;
    font-weight: 600;
}

.search-form__btn-wrapper {
    display: flex;
    align-content: center;
    justify-content: flex-start;
    align-items: baseline;
}

.search-form__btn-wrapper button:last-child {
    margin-left: auto;
}