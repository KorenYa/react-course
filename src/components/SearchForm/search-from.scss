@import '../../styles/constants.scss';

.search-form {
    padding: 20px 60px;
    background: url('../../../src/img/bg.jpg') center center;
    color: $white;

    &__title {
        color: $red;
        font-weight: $bold;
        margin-bottom: 30px;
    }

    &__subtitle {
        text-transform: uppercase;
        font-size: $md;
        font-weight: $bold;
    }

    &__btn-wrapper {
        display: flex;
        align-content: center;
        justify-content: flex-start;
        align-items: baseline;

        button:last-child {
            margin-left: auto;
        }
    }
}
