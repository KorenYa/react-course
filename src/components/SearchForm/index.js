import React, { Component } from 'react';
import Title from '../Title';
import Input from '../Input';
import Button from '../Button';
import noop from '../../utils/noop';
import PropTypes from 'prop-types';
import './search-from.css';
const SearchForm = function(props) {
    return (
        <form className='search-form' onSubmit={props.handleSubmit}>
            <div className='search-form__title'>netflixroulette</div>
            <Title title='find your movie' />
            <Input value={props.inputValue} handleChange={props.handleChange} />
            <div className='search-form__btn-wrapper'>
                <span className='search-form__subtitle'>Search by</span>
                <Button
                    type='button'
                    text='Title'
                    searchBy='title'
                    onClick={props.handleFilter}
                    styleName={props.searchBy == 'title' ? 'active' : ''}
                />
                <Button
                    type='button'
                    text='Genre'
                    searchBy='genre'
                    onClick={props.handleFilter}
                    styleName={props.searchBy == 'genre' ? 'active' : ''}
                />
                <Button
                    type='submit'
                    text={'Search'}
                    styleName='btn-lg'
                    style={{ alignSelf: 'flex-end' }}
                    onClick={noop}
                />
            </div>
        </form>
    );
};
SearchForm.propTypes = {
    handleSubmit: PropTypes.func,
    handleChange: PropTypes.func,
    handleFilter: PropTypes.func,
    inputValue: PropTypes.string,
    searchBy: PropTypes.string
};
export default SearchForm;
