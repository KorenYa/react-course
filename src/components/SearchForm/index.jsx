import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Title from '../Title';
import Input from '../Input';
import Button from '../Button';
import noop from '../../utils/noop';
import './search-from.scss';

const SearchForm = (props) => {
  const {
    handleSubmit,
    handleChange,
    handleFilter,
    searchBy,
    inputValue,
  } = props;
  return (
    <form className="search-form" onSubmit={handleSubmit}>
      <div className="search-form__title">netflixroulette</div>
      <Title title="find your movie" />
      <Input value={inputValue} handleChange={handleChange} />
      <div className="search-form__btn-wrapper">
        <span className="search-form__subtitle">Search by</span>
        <Button
          type="button"
          text="Title"
          searchBy="title"
          onClick={handleFilter}
          styleName={searchBy === 'title' ? 'active' : ''}
        />
        <Button
          type="button"
          text="Genre"
          searchBy="genre"
          onClick={handleFilter}
          styleName={searchBy === 'genre' ? 'active' : ''}
        />
        <Button
          type="submit"
          text="Search"
          styleName="btn-lg"
          style={{ alignSelf: 'flex-end' }}
          onClick={noop}
        />
      </div>
    </form>
  );
};
SearchForm.propTypes = {
  handleSubmit: PropTypes.func,
  handleChange: PropTypes.func,
  handleFilter: PropTypes.func,
  inputValue: PropTypes.string,
  searchBy: PropTypes.string,
};
export default SearchForm;
