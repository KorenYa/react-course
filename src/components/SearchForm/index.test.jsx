import React from 'react';
import renderer from 'react-test-renderer';

import SearchForm from '.';

describe('SearchForm', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <SearchForm
          handleSubmit={jest.fn()}
          handleChange={jest.fn()}
          handleFilter={jest.fn()}
          inputValue="Narnia"
          searchBy="title"
        />,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
