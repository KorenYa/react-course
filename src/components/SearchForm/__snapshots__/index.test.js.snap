// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm renders correctly 1`] = `
<form
  className="search-form"
  onSubmit="() => {}"
>
  <div
    className="search-form__title"
  >
    netflixroulette
  </div>
  <h1
    className="title"
  >
    find your movie
  </h1>
  <input
    className="input"
    onChange="() => {}"
    type="text"
    value="Narnia"
  />
  <div
    className="search-form__btn-wrapper"
  >
    <span
      className="search-form__subtitle"
    >
      Search by
    </span>
    <button
      className="btn active"
      onClick={[Function]}
      type="button"
    >
      Title
    </button>
    <button
      className="btn "
      onClick={[Function]}
      type="button"
    >
      Genre
    </button>
    <button
      className="btn btn-lg"
      onClick={[Function]}
      type="submit"
    >
      Search
    </button>
  </div>
</form>
`;
